### PR TITLE
[Behat] Add context aliases on interfaces

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Branch?         | 1.4
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

This allows to use autowiring on your custom behat contexts when customizing Behat features of your project.

### CODE REVIEW

**Comment by loic425** [comment]
@pamil @lchrusciel I just cherry picked an old POC. Maybe It needs some refreshments.

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 use Behat\Behat\Context\Context;
 use Sylius\Behat\Page\Admin\Crud\IndexPageInterface;
+use Sylius\Behat\Page\Admin\Customer as Customer;
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php]: Something went wrong with this refactor

File: src/Sylius/Behat/Resources/config/services/pages/admin/account.xml [hunk 2]
```diff
@@ -13,8 +13,10 @@
 
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
-        <defaults public="true" />
+        <defaults public="false" />
```
Comment (lchrusciel) [inline src/Sylius/Behat/Resources/config/services/pages/admin/account.xml]: Do we want to make all the behat services private? /cc @pamil

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php (hunk 1 above)
Comment (loic425) [reply src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php]: Yes, I think we can use CustomerIndexPageInterface on constructor arguments..

File: src/Sylius/Behat/Resources/config/services/pages/admin/account.xml (hunk 2 above)
Comment (loic425) [reply src/Sylius/Behat/Resources/config/services/pages/admin/account.xml]: I propose that for all the pages, not for the Behat context services. 
All services which implements Behat\Behat\Context\Context needs to be public.
Moreover, friends-of-behat/symfony-extension with autoconfigure make them public.

Comment (loic425) [reply src/Sylius/Behat/Resources/config/services/pages/admin/account.xml]: @lchrusciel Most of them are already private, overriding the file defaults configuration:
- src/Sylius/Behat/Resources/config/services/pages/admin/admin_user.xml
- src/Sylius/Behat/Resources/config/services/pages/admin/channel.xml
- src/Sylius/Behat/Resources/config/services/pages/admin/country.xml

**Comment by loic425** [comment]
@pamil @lchrusciel 
I made a big refesh of this pull-request. This is the first step to use autowiring on Behat contexts.
This would be awesome for Full Sylius E-commerce developers which wants to add their Behat suites.

**Review by lchrusciel — APPROVED** [review]

**Comment by loic425** [comment]
@pamil @lchrusciel do not stable this please :)

**Comment by lchrusciel** [comment]
Hey Loic,

are you open to bringing life back to this PR? Perhaps we could merge it to the master

**Comment by jakubtobiasz** [comment]
@loic425 AFAIR you've created two new PRs replacing this one, right? Can we close it?

**Pull request closed without merging**